lib/chef/provider/windows_task.rb

```diff
@@ -201,308 +224,303 @@ def action_disable
-            task[:day] << e.to_s[0..3].delete("<").delete("/>")
+      def months_of_year
+        months_of_year = []
+        if new_resource.months
+          months = new_resource.months.split(',').map(&:upcase)
+          if months.size > 1
+            months.each do |month|
+              month = month.strip.to_sym
+              months_of_year << MONTHS[month]
+            end
+            months_of_year = months_of_year.inject(:|)
+          else
+            months_of_year = MONTHS[new_resource.months.upcase.to_sym]
           end
-          task[:day] = task[:day].join(", ")
-        end
-
-        # for frequency = :monthly
-        task[:schedule_by_month] = root.elements["Triggers/CalendarTrigger/ScheduleByMonth/DaysOfMonth/Day"].text if root.elements["Triggers/CalendarTrigger/ScheduleByMonth/DaysOfMonth/Day"]
-        if root.elements["Triggers/CalendarTrigger/ScheduleByMonth/Months"]
-          task[:months] = []
-          root.elements["Triggers/CalendarTrigger/ScheduleByMonth/Months"].elements.each do |e|
-            task[:months] << e.to_s[0..3].delete("<").delete("/>")
+        else
+          months = MONTHS.each do |key, value|
+            months_of_year << MONTHS[key]
           end
-          task[:months] = task[:months].join(", ")
+          months_of_year = months_of_year.inject(:|)
         end
+        months_of_year
+      end
 
-        task[:on_logon] = true if root.elements["Triggers/LogonTrigger"]
-        task[:onstart] = true if root.elements["Triggers/BootTrigger"]
-        task[:on_idle] = true if root.elements["Triggers/IdleTrigger"]
-
-        task[:idle_time] = root.elements["Settings/IdleSettings/Duration"].text if root.elements["Settings/IdleSettings/Duration"] && task[:on_idle]
+      # TODO handle run_level here
+      def run_level
+        case new_resource.run_level
+        when :highest
+          TaskScheduler::TASK_RUNLEVEL_HIGHEST
+        when :limited
+          TaskScheduler::TASK_RUNLEVEL_LUA
+        end
+      end
 
-        task[:none] = true if root.elements["Triggers/"] && root.elements["Triggers/"].entries.empty?
-        task[:once] = true if !(task[:repetition_interval] || task[:schedule_by_day] || task[:schedule_by_week] || task[:schedule_by_month] || task[:on_logon] || task[:onstart] || task[:on_idle] || task[:none])
-        task[:execution_time_limit] = root.elements["Settings/ExecutionTimeLimit"].text if root.elements["Settings/ExecutionTimeLimit"] #by default PT72H
-        task[:random_delay] = root.elements["Triggers/TimeTrigger/RandomDelay"].text if root.elements["Triggers/TimeTrigger/RandomDelay"]
-        task[:random_delay] = root.elements["Triggers/CalendarTrigger/RandomDelay"].text if root.elements["Triggers/CalendarTrigger/RandomDelay"]
-        task
+      #TODO: while creating the configuration settings win32-taskscheduler it accepts execution time limit values in ISO8601 formata
+      def config_settings
+        settings = {}
+        settings = {
```

Comment:
pls use similar format.
```
settings = {}
settings[:execution_time_limit] = ISO8601::Duration.new(new_resource.execution_time_limit.to_i).to_s
settings[:enabled] = true
```
if we are using something like this, then above review comment can be ignored(of declaration)